pkg/kubectl/cmd/set/set_subject.go

```diff
@@ -0,0 +1,264 @@
+	}
+
+	resourceTypesWithPodTemplate := []string{}
+	for _, resource := range f.SuggestedPodTemplateResources() {
+		resourceTypesWithPodTemplate = append(resourceTypesWithPodTemplate, resource.Resource)
+	}
+
+	cmd := &cobra.Command{
+		Use:     "subject (-f FILENAME | TYPE NAME) [--user=username] [--group=groupname] [--serviceaccount=namespace:serviceaccountname] [--dry-run]",
+		Short:   i18n.T("Update User, Group or ServiceAccount in a RoleBinding/ClusterRoleBinding"),
+		Long:    subject_long,
+		Example: subject_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run(addSubjects))
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "the resource to update the subjects"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().BoolVar(&options.All, "all", false, "select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set resources will NOT contact api-server but run locally.")
+	cmdutil.AddDryRunFlag(cmd)
+	cmd.Flags().StringArrayVar(&options.Users, "user", []string{}, "usernames to bind to the role")
+	cmd.Flags().StringArrayVar(&options.Groups, "group", []string{}, "groups to bind to the role")
+	cmd.Flags().StringArrayVar(&options.Serviceaccounts, "serviceaccount", []string{}, "service accounts to bind to the role")
+	return cmd
+}
+
+func (o *SubjectOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Local = cmdutil.GetFlagBool(cmd, "local")
+	o.Mapper, o.Typer = f.Object()
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
+	o.PrintObject = f.PrintObject
+	o.Cmd = cmd
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *SubjectOptions) Validate() error {
+	if len(o.Users) == 0 && len(o.Groups) == 0 && len(o.Serviceaccounts) == 0 {
+		return fmt.Errorf("you must specify at least one value of user, group or serviceaccount")
+	}
+
+	for _, sa := range o.Serviceaccounts {
+		tokens := strings.Split(sa, ":")
+		if len(tokens) != 2 || tokens[0] == "" || tokens[1] == "" {
+			return fmt.Errorf("serviceaccount must be <namespace>:<name>")
```

Comment:
TODO: I'd like to find a way to allow `:name` for `roles`, but not `clusterroles`.  For roles it indicates, "use the current namespace"